compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -1029,6 +1037,16 @@ fn llvm_fixup_input<'ll, 'tcx>(
                 _ => value,
             }
         }
+        (InlineAsmRegClass::RiscV(RiscVInlineAsmRegClass::freg), Abi::Scalar(s))
+            if s.primitive() == Primitive::Float(Float::F16)
+                && !bx.tcx.asm_target_features(instance.def_id()).contains(&sym::zfhmin) =>
```

Comment:
This needs to check for zfh as well. We currently won't automatically enable zfhmin if you only enable zfh for a function with `#[target_feature]` because rustc isn't aware of feature dependencies.